lib/SILOptimizer/Transforms/SILMem2Reg.cpp

```diff
@@ -1164,7 +1178,8 @@ SILInstruction *StackAllocationPromoter::promoteAllocationInBlock(
                      /*isStorageValid=*/true};
       // The current store is now the lastStoreInst.
       lastStoreInst = sbi;
-      if (lexicalLifetimeEnsured(asi)) {
+      if (lexicalLifetimeEnsured(asi) &&
```

Comment:
Should we add a predicate that combines these two conditions, considering how many times it's showing up?